### PR TITLE
[DOCS] minor fix to documentation: http.host can't default to itself

### PR DESCRIPTION
**Pull request by SylvainJuge**
- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

File: docs/reference/modules/http.asciidoc [hunk 1]
```diff
@@ -36,7 +36,7 @@ from the outside. Defaults to the actual port assigned via `http.port`.
 
 |`http.publish_host` |The host address to publish for HTTP clients to connect to. Defaults to `http.host` (if set) or `network.publish_host`.
 
-|`http.host` |Used to set the `http.bind_host` and the `http.publish_host` Defaults to `http.host` or `network.host`.
+|`http.host` |Used to set the `http.bind_host` and the `http.publish_host` Defaults to `network.host`.
```
Comment (DaveCTurner) [inline docs/reference/modules/http.asciidoc]: Technically it defaults to an empty list, but it only has an effect if set (see previous two settings) so I don't think we need to talk about the default at all:

```suggestion
|`http.host` |Used to set the `http.bind_host` and the `http.publish_host`.
```

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
I suggested deleting the mention of the default entirely -- see below.

There is a similar bug in the docs for `transport.host`. Would you fix that too?

**Comment by SylvainJuge** [comment]
@DaveCTurner I've applied the same change for `transport.host`.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by SylvainJuge** [comment]
[backport to 7.4](https://github.com/elastic/elasticsearch/commit/43f91a3e8628ad9289beb3ccf71b4959c042e3be)

[backport to 7.5](https://github.com/elastic/elasticsearch/commit/d116f827cab1617b4d6a689918fa70bf05845972)

[backport to 7.x](https://github.com/elastic/elasticsearch/commit/e8f49cdee06e1faf4c8ba95269c6384ac2ff333d)

**Pull request merged**